drivers/interrupt_controller/intc_plic.c

```diff
@@ -29,7 +29,7 @@
 #define PLIC_IRQS        (CONFIG_NUM_IRQS - CONFIG_2ND_LVL_ISR_TBL_OFFSET)
 #define PLIC_EN_SIZE     ((PLIC_IRQS >> 5) + 1)
 
-#define PLIC_EDGE_TRIG_TYPE   (PLIC_MAX_PRIO + DT_INST_PROP(0, riscv_trigger_reg_offset))
+#define PLIC_EDGE_TRIG_TYPE   DT_INST_PROP(0, riscv_trigger_reg_offset)
```

Comment:
After looking at it in detail, I wondered how this even passed CI in the first place, and it seems to be that there is zero test coverage at the moment.

Good in a way (nothing was broken), but also bad in a way (interrupt controller code should be thoroughly tested).

@ycsin - please file a bug report, because this will need a backport to v3.4-branch and also please fix `riscv_plic_is_edge_irq()` in `drivers/interrupt_controller/intc_plic.c`.

IIRC, the correct version of that would be something like this:
```cpp
static int riscv_plic_is_edge_irq(uint32_t irq)
{
	if (IS_ENABLED(CONFIG_PLIC_SUPPORTS_EDGE_IRQ)) {
		volatile uint32_t *trig = (volatile uint32_t *)PLIC_REG + PLIC_EDGE_TRIG_TYPE;

		trig += (irq >> PLIC_EDGE_TRIG_SHIFT);
		return *trig & BIT(irq);
	}
	return 0;
}
```

Reply:
I think this is a great example of why PLIC should be multi-instance. With that, we can add it to a `build_all` test relatively easily and permute config options in `testcase.yaml`. But multi-instance should come later.

And, honestly, even for drivers that are never intended to be multi-instance, I always recommend that devs do so anyway in order to avoid hard-coding instance numbers in macros, use proper data encapsulation, etc.

Reply:
> And, honestly, even for drivers that are never intended to be multi-instance, I always recommend that devs do so anyway in order to avoid hard-coding instance numbers in macros, use proper data encapsulation, etc.

+1